jerry-core/ecma/builtin-objects/ecma-builtin-number-prototype.c

```diff
@@ -574,266 +472,301 @@ typedef enum
-                                                      : LIT_MAGIC_STRING_INFINITY_UL));
+    if (!is_zero)
+    {
+      ecma_stringbuilder_append_char (&builder, LIT_CHAR_MINUS);
+    }
+
+    this_num *= -1;
   }
 
-  if (mode == NUMBER_ROUTINE_TO_PRECISION &&
-      (ecma_number_is_nan (arg_num) || arg_num < 1 || arg_num >= 22))
+  /* Handle zero separately */
+  if (is_zero)
   {
-    return ecma_raise_range_error (ECMA_ERR_MSG ("Precision digits must be between 1 and 21."));
-  }
+    if (mode == NUMBER_ROUTINE_TO_PRECISION)
+    {
+      arg_int--;
+    }
 
-  *arg_1_int32_p = ecma_number_to_int32 (arg_num);
+    ecma_stringbuilder_append_char (&builder, LIT_CHAR_0);
 
-  return ECMA_VALUE_EMPTY;
-} /* ecma_builtin_number_prepare_conversion */
+    if (arg_int > 0)
+    {
+      ecma_stringbuilder_append_char (&builder, LIT_CHAR_DOT);
+    }
 
-/**
- * The Number.prototype object's 'toFixed' routine
- *
- * See also:
- *          ECMA-262 v5, 15.7.4.5
- *
- * @return ecma value
- *         Returned value must be freed with ecma_free_value.
- */
-static ecma_value_t
-ecma_builtin_number_prototype_object_to_fixed (ecma_number_t this_num, /**< this argument number */
-                                               ecma_value_t radix) /**< routine's argument */
-{
-  bool is_negative;
-  int32_t frac_digits;
+    for (int32_t i = 0; i < arg_int; i++)
+    {
+      ecma_stringbuilder_append_char (&builder, LIT_CHAR_0);
+    }
+
+    if (mode == NUMBER_ROUTINE_TO_EXPONENTIAL)
+    {
+      ecma_stringbuilder_append_raw (&builder, (const lit_utf8_byte_t *) "e+0", 3);
+    }
 
-  ecma_value_t comp_value = ecma_builtin_number_prepare_conversion (&this_num,
-                                                                    radix,
-                                                                    &is_negative,
-                                                                    &frac_digits,
-                                                                    NUMBER_ROUTINE_TO_FIXED);
+    return ecma_make_string_value (ecma_stringbuilder_finalize (&builder));
+  }
 
-  if (!ecma_is_value_empty (comp_value))
+  /* Handle infinity separately */
+  if (ecma_number_is_infinity (this_num))
   {
-    return comp_value;
+    ecma_stringbuilder_append_magic (&builder, LIT_MAGIC_STRING_INFINITY_UL);
+    return ecma_make_string_value (ecma_stringbuilder_finalize (&builder));
   }
 
-  /* Get the parameters of the number if non-zero. */
-  lit_utf8_byte_t digits[ECMA_MAX_CHARS_IN_STRINGIFIED_NUMBER];
-  lit_utf8_size_t num_digits;
-  int32_t exponent;
+  num_of_digits = ecma_number_to_decimal (this_num, digits, &exponent);
 
-  if (!ecma_number_is_zero (this_num))
-  {
-    num_digits = ecma_number_to_binary_floating_point_number (this_num, digits, &exponent);
-    JERRY_ASSERT (exponent >= 0);
-  }
-  else
+  /* Handle undefined argument */
+  if (ecma_is_value_undefined (arg))
```

Comment:
The required actions when the argument is undefined also changed slightly n the standard, please double check them.